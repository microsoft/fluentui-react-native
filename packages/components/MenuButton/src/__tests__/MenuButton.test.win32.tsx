import * as React from 'react';
import { ContextualMenuItemProps } from '@fluentui-react-native/contextual-menu';
import { MenuButton } from '..';
import * as renderer from 'react-test-renderer';

it('ContextualMenu default props', () => {
  const menuItems: ContextualMenuItemProps[] = [
    {
      itemKey: '1',
      text: 'MenuItem 1',
    },
    {
      itemKey: '2',
      text: 'MenuItem 2',
    },
    {
      itemKey: '3',
      text: 'MenuItem 3',
      disabled: true,
    },
  ];
  const tree = renderer.create(<MenuButton content="Standard MenuButton" menuItems={menuItems} />).toJSON();
  expect(tree).toMatchSnapshot();
});
