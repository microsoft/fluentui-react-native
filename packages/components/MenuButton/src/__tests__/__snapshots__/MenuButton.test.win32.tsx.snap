// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextualMenu default props 1`] = `
<View
  accessible={false}
  focusable={false}
  style={
    Object {
      "alignItems": "flex-start",
      "alignSelf": "flex-start",
      "backgroundColor": "#f3f2f1",
      "borderColor": "#8a8886",
      "borderRadius": 12,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "overflow": "hidden",
    }
  }
>
  <View
    accessibilityLabel="Standard MenuButton"
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    componentRef={
      Object {
        "current": null,
      }
    }
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "minHeight": 52,
          "minWidth": 80,
          "paddingHorizontal": 20,
          "paddingVertical": 14,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#323130",
            "fontFamily": "Segoe UI",
            "fontSize": 14,
            "fontWeight": "600",
            "margin": 0,
          }
        }
      >
        Standard MenuButton
      </Text>
    </View>
  </View>
</View>
`;
